### PR TITLE
Update program/commander & add space captions download

### PR DESCRIPTION
**Pull request by HitomaruKonpaku**

### CODE REVIEW

File: src/SpaceCaptionsDownloader.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  private count = 0
+  private cursor = ''
+
+  constructor(
+    private spaceId: string,
+    private accessToken: string,
+    private file?: string,
+  ) {
+    this.logger = baseLogger.child({ label: `[SpaceCaptionsDownloader@${spaceId}]` })
+    this.file = this.file || `${Date.now()}.jsonl`
```
Comment (IAmVisco) [inline src/SpaceCaptionsDownloader.ts:20]: Just date is not enough. See if you can get id, name etc.

File: src/SpaceCaptionsDownloader.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      fs.writeFileSync(this.file, '')
+      do {
+        this.count += 1
+        this.logger.info(`Downloading part ${this.count}`)
+        // eslint-disable-next-line no-await-in-loop
+        const history = await this.getChatHistory()
+        const messages = Array.from(history.messages)
+        messages.forEach((message) => {
+          const payload = `${JSON.stringify(message)}\n`
+          fs.appendFileSync(this.file, payload)
```
Comment (IAmVisco) [inline src/SpaceCaptionsDownloader.ts:35]: Can be shortened into one liner I guess.

File: src/SpaceCaptionsDownloader.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        })
+        this.cursor = history.cursor
+      } while (this.cursor)
+    } catch (error) {
+      this.logger.error(error.message)
+    }
+  }
+
+  private async getChatHistory() {
+    const { data } = await axios.post(this.API_URL, {
```
Comment (IAmVisco) [inline src/SpaceCaptionsDownloader.ts:45]: Types doko

File: src/SpaceWatcher.ts [hunk 4]
```diff
@@ -52,13 +54,16 @@ export class SpaceWatcher extends EventEmitter {
       this.showNotification()
       this.mediaKey = this.metadata.media_key
       this.liveStreamStatus = await Util.getLiveVideoStreamStatus(this.mediaKey, headers)
+      this.logger.debug('liveStreamStatus', this.liveStreamStatus)
+      this.accessChatData = await Util.getAccessChatData(this.liveStreamStatus.chatToken)
```
Comment (IAmVisco) [inline src/SpaceWatcher.ts:58]: Is it really util?

File: src/Util.ts [hunk 5]
```diff
@@ -111,6 +112,14 @@ export class Util {
     return data
   }
 
+  public static async getAccessChatData(chatToken: string) {
+    const { data } = await axios.post<AccessChat>(
```
Comment (IAmVisco) [inline src/Util.ts:116]: Is it `AccessChat` or just `Chat` that you access?

File: src/index.ts [hunk 6]
```diff
@@ -1,50 +1,65 @@
...
-      }
+  const { url } = args
+  if (url) {
+    logger.info('Starting in url mode', { url })
+    await Downloader.downloadSpace(url, Util.getTimeString())
+    return
+  }
+
+  const main = new Main()
+  const { id } = args
```
Comment (IAmVisco) [inline src/index.ts:44]: You might wanna defined them all at once and then just (cleaner?)

**Review by IAmVisco — CHANGES_REQUESTED** [review]

File: src/Util.ts (hunk 5 above)
Comment (HitomaruKonpaku) [reply src/Util.ts:116]: It includes `access_token`, `endpoint` and some other info to access the chat room

File: src/SpaceWatcher.ts (hunk 4 above)
Comment (HitomaruKonpaku) [reply src/SpaceWatcher.ts:58]: Plan to reuse it so I put it there 🤔

**Review by IAmVisco — APPROVED** [review]

**Pull request merged**